evm/15_core-concepts/49_common-erc-standards.md

```diff
@@ -0,0 +1,101 @@
+    }
+
+    function balanceOf(address account) external view returns (uint256) {
+        return balances[account];
+    }
+
+    function transfer(address recipient, uint256 amount) external returns (bool) {
+        require(amount <= balances[msg.sender], "Insufficient balance");
+        balances[msg.sender] -= amount;
+        balances[recipient] += amount;
+        return true;
+    }
+
+    function approve(address spender, uint256 amount) external returns (bool) {
+        allowances[msg.sender][spender] = amount;
+        return true;
+    }
+
+    function transferFrom(address sender, address recipient, uint256 amount) external returns (bool) {
+        require(amount <= balances[sender], "Insufficient balance");
+        require(amount <= allowances[sender][msg.sender], "Insufficient allowance");
+        balances[sender] -= amount;
+        balances[recipient] += amount;
+        allowances[sender][msg.sender] -= amount;
+        return true;
+    }
+}
+```
+
+
+### ERC-721
+ERC-721 is a standard for non-fungible tokens (NFTs) on the EOSEVM network. NFTs are unique and indivisible tokens that represent ownership of a specific item or asset. Each NFT has a distinct identifier, and they can be used for digital collectibles, unique assets, and more.
+
+#### Example ERC-721 Contract
+Here's an example of a basic ERC-721 token contract using the OpenZeppelin library in Solidity:
+
+```
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+contract MyNFT is ERC721 {
+    using Counters for Counters.Counter;
+    Counters.Counter private _tokenIds;
+
+    constructor() ERC721("MyNFT", "NFT") {}
+
+    function mintNFT(address recipient, string memory tokenURI) external returns (uint256) {
+        _tokenIds.increment();
+        uint256 newItemId = _tokenIds.current();
+        _safeMint(recipient, newItemId);
+        _setTokenURI(newItemId, tokenURI);
+        return newItemId;
+    }
+}
+```
+
+### ERC-777
+ERC-777 is an improved fungible token standard that incorporates additional features compared to ERC-20. It introduces advanced functionalities like hooks, which allow contracts to react to token transfers, and operator approvals, which enable third-party operators to handle tokens on behalf of token holders.
+
+### ERC-1155
+ERC-1155 is a multi-token standard that enables the creation of both fungible and non-fungible tokens within a single contract. This standard is particularly useful for projects that require the management of multiple types of tokens, such as gaming platforms or decentralized exchanges.
+
+These ERC standards provide a foundation for building various token-based applications and use cases on the EOSEVM network. It's important to note that while these standards originated on Ethereum, they can also be implemented and utilized within the EOSEVM ecosystem.
```

Comment:
Looks like this paragraph is still under the 1155 section (table of contents-wise), but doesn't read like it is. Maybe add a `## Final thoughts` or something?